Assets/Scripts/Actions/CreateShipyard.cs

```diff
@@ -3,15 +3,16 @@
 using Actions;
 using UnityEngine;
 
-public class CreateShipyard : MonoBehaviour, IAction
-{
+public class CreateShipyard : MonoBehaviour, IAction {
     public GameObject shipyard;
 
-    public void execute()
-    {
+    public void execute() {
         var position = transform.position;
         Instantiate(shipyard, new Vector3(position.x + 1, position.y - 1.5f, position.z + 1),
             Quaternion.identity);
         Destroy(this.gameObject);
+
+        var gameInterface = FindObjectOfType<GameUI>();
```

Comment:
Since we only have 1 GameUI i think it's safer to simply pass it as a public variable, finding objects is always a bit shady imo (doesn't really matter too much).

Also this shouldn't be called when creating the shipyard but when the tier 4 boat is built, there is a message "game should end" or something I believe